applications/asset_tracker_v2/tests/lwm2m_codec/src/lwm2m_codec_test.c

```diff
@@ -0,0 +1,769 @@
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, GNSS_TIMEOUT_RID),
+		cfg.gnss_timeout, 0);
+
+	__wrap_lwm2m_engine_set_s32_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, ACTIVE_WAIT_TIMEOUT_RID),
+		cfg.active_wait_timeout, 0);
+
+	__wrap_lwm2m_engine_set_s32_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, MOVEMENT_RESOLUTION_RID),
+		cfg.movement_resolution, 0);
+
+	__wrap_lwm2m_engine_set_s32_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, MOVEMENT_TIMEOUT_RID),
+		cfg.movement_timeout, 0);
+
+	__wrap_lwm2m_engine_set_bool_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, PASSIVE_MODE_RID),
+		!cfg.active_mode, 0);
+
+	__wrap_lwm2m_engine_set_bool_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, GNSS_ENABLE_RID),
+		!cfg.no_data.gnss, 0);
+
+	__wrap_lwm2m_engine_set_bool_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, NEIGHBOR_CELL_ENABLE_RID),
+		!cfg.no_data.neighbor_cell, 0);
+
+	/* Register callbacks from changes in device configurations. */
+	__wrap_lwm2m_engine_register_post_write_callback_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, PASSIVE_MODE_RID),
+		NULL, 0);
+	__wrap_lwm2m_engine_register_post_write_callback_IgnoreArg_cb();
+
+	__wrap_lwm2m_engine_register_post_write_callback_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, GNSS_TIMEOUT_RID),
+		NULL, 0);
+	__wrap_lwm2m_engine_register_post_write_callback_IgnoreArg_cb();
+
+	__wrap_lwm2m_engine_register_post_write_callback_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, ACTIVE_WAIT_TIMEOUT_RID),
+		NULL, 0);
+	__wrap_lwm2m_engine_register_post_write_callback_IgnoreArg_cb();
+
+	__wrap_lwm2m_engine_register_post_write_callback_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, MOVEMENT_RESOLUTION_RID),
+		NULL, 0);
+	__wrap_lwm2m_engine_register_post_write_callback_IgnoreArg_cb();
+
+	__wrap_lwm2m_engine_register_post_write_callback_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, MOVEMENT_TIMEOUT_RID),
+		NULL, 0);
+	__wrap_lwm2m_engine_register_post_write_callback_IgnoreArg_cb();
+
+	__wrap_lwm2m_engine_register_post_write_callback_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, ACCELEROMETER_THRESHOLD_RID),
+		NULL, 0);
+	__wrap_lwm2m_engine_register_post_write_callback_IgnoreArg_cb();
+
+	__wrap_lwm2m_engine_register_post_write_callback_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, GNSS_ENABLE_RID),
+		NULL, 0);
+	__wrap_lwm2m_engine_register_post_write_callback_IgnoreArg_cb();
+
+	__wrap_lwm2m_engine_register_post_write_callback_ExpectAndReturn(
+		LWM2M_PATH(CONFIGURATION_OBJECT_ID, 0, NEIGHBOR_CELL_ENABLE_RID),
+		NULL, 0);
+	__wrap_lwm2m_engine_register_post_write_callback_IgnoreArg_cb();
+
+	__wrap_lwm2m_engine_register_post_write_callback_AddCallback(&post_write_callback_stub);
+
+	TEST_ASSERT_EQUAL(0, cloud_codec_init(&cfg, cloud_codec_event_handler));
+}
+
+void test_lwm2m_cloud_codec_encode_neighbor_cells(void)
```

Comment:
In the tests starting from here, dont we need to call `cloud_codec_init()` as a pre-condition? Or thats not needed for the encode APIs to work?

Reply:
Yes, that would be necessary for some of these that depend on certain resource buffers being initiated prior to the call. In actual code, the lwm2m engine would definitely complain if this was not the case.